zcash_client_sqlite/src/wallet.rs

```diff
@@ -809,6 +816,27 @@ pub(crate) fn truncate_to_height<P: consensus::Parameters>(
             "DELETE FROM blocks WHERE height > ?",
             [u32::from(block_height)],
         )?;
+
+        // Delete from the scanning queue any range with a start height greater than or equal to
+        // the truncation height, and truncate any remaining range by setting the end equal to
+        // the truncation height.
+        conn.execute(
+            "DELETE FROM scan_queue
+            WHERE block_range_start >= :block_height",
+            named_params![":block_height": u32::from(block_height)],
+        )?;
+
+        conn.execute(
+            "UPDATE scan_queue
+            SET block_range_end = :block_height
+            WHERE block_range_end > :block_height",
+            named_params![":block_height": u32::from(block_height)],
+        )?;
```

Comment:
If the `block_range_end` column is exclusive, then this should be:
```suggestion
        // Delete from the scanning queue any range with a start height greater than
        // the truncation height, and truncate any remaining range by setting the end
        // equal to the truncation height + 1.
        conn.execute(
            "DELETE FROM scan_queue
            WHERE block_range_start > :block_height",
            named_params![":block_height": u32::from(block_height)],
        )?;

        conn.execute(
            "UPDATE scan_queue
            SET block_range_end = :end_height
            WHERE block_range_end > :end_height",
            named_params![":end_height": u32::from(block_height + 1)],
        )?;
```